taskcluster/client.py

```diff
@@ -490,6 +490,13 @@ def _makeHttpRequest(self, method, route, payload):
                     "Failed to establish connection",
                     superExc=rerr
                 )
+                
+            # Handles Non JSON responses
+            answer = response.headers['Content-Type']
+            answer = answer.strip()
+            contentType = "application/json"
+            if answer != contentType:
+                return response.raw
```

Comment:
I didn't know that they exposed a `.raw` property here, but I think that returning the requests-library response (i.e. `return response`) should be enough.

Reply:
We also need to make this choice later on.  The reason is that we haven't yet classified the response as an error or a successful response.  If we did this here, we'd never be checking for non-200 series responses.  This logic should probably go near the existing line `return response.json()`.